cibuild.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -e -x
+
+WD=$PWD
+nuttx=$WD/../nuttx
+apps=$WD/../apps
+tools=$WD/../tools
+prebuilt=$WD/../prebuilt
+
+install="gen-romfs kconfig-frontends arm-gcc-toolchain mips-gcc-toolchain riscv-gcc-toolchain c-cache"
```

Comment:
A minor comment: I think that maybe these names are ok with the dash '-', since they look as option parameters, but since they are used as function names, I am not sure, since think that using '-' instead of '_' is less common. I haven't find any NuttX shell script coding style, so maybe it is ok to leave it like that. Also, I don't see why the 'c-cache' is spelled that way instead of simply 'ccache', since that seems to be the name of the tool.

Reply:
Using  '-' instead of '_' is mainly align to the names of install tools. But genromfs and ccache functions names would conflict with the genromfs/ccache commands names. So I have to rename genromfs -> gen-romfs, ccache -> c-cache to avoid run abnormally : (